erpnext/hr/doctype/training_event/training_event.py

```diff
@@ -8,20 +8,19 @@
 from frappe.model.document import Document
 
 class TrainingEvent(Document):
-	def validate(self):
-		if self.event_status == "Scheduled":
-			self.invite_employee()
+	def on_update(self):
+		self.invite_employee()
 	
 	def on_update_after_submit(self):
-		if self.event_status == "Scheduled" and self.send_email:
-			self.invite_employee()
+		self.invite_employee()
 	
 	def invite_employee(self):
-		subject = _("""You are invited for to attend {0} - {1} scheduled from {2} to {3} at {4}."""\
-			.format(self.type, self.event_name, self.start_time, self.end_time, self.location))
-	
-		for emp in self.employees:
-			if emp.status== "Open":
-				frappe.sendmail(frappe.db.get_value("Employee", emp.employee, "company_email"), \
-					subject=subject, content= self.introduction)
-				emp.status= "Invited"
\ No newline at end of file
+		if self.event_status == "Scheduled" and self.send_email:
+			subject = _("""You are invited for to attend {0} - {1} scheduled from {2} to {3} at {4}."""\
```

Comment:
Lets do email alert for this? Its a much more elegant solution since the user can design the content of the email. You can add it to the fixtures and then call it when `send_email` is true.